### DISCUSSION TITLE
How to specify the color of lines and bars in XYChart?

### DISCUSSION THREAD

**Discussion by jeanplevesque** [mermaid-js/mermaid#5207] [Q&A] [answered]
I looked at the documentation and I can't figure out how to set the color for `bar` and `line` definitions. My use-case is having multiple lines in the same chart and create a legend to explain the meaning of each line.

**Comment by eadmaster** [comment]
i think mermaid should auto-pick different web-safe colors for xygraph bars when no theme is specified.

This should make diagrams looks better. e.g.:
![graph](https://github.com/mermaid-js/mermaid/assets/925171/8741b79d-324c-485f-b112-deea755066fd)

compare with the [example here](https://mermaid.js.org/syntax/xyChart.html#example-on-config-and-theme)

**Comment by Piratepie** [comment, marked as answer]
Using the [Chart Theme Variables](https://mermaid.js.org/syntax/xyChart.html#chart-theme-variables) you can set a color palette with `plotColorPalette`.
These colors will be applied to the lines and bars defined in the xyChart. So the first color in your palette will correspond with the first bar/line etc.

Here's a short example 

```
---
config:
    themeVariables:
        xyChart:
            plotColorPalette: "#000000, #0000FF, #00FF00, #FF0000"
---
         %% plotColorPalette: "black, blue, green, red"
xychart-beta
    title "Different colors in xyChart"
    x-axis "numbersX" ["Group A", "Group B", "Group C", "Group D"]
    y-axis "numbersY" 0 --> 40
    %% black
    bar [20,30,25,30]
    %% blue
    bar [10,20,20,20]
    %% green
    bar [5,10,5,5]
    %% red
    line [0,5,40,2]
```
![example](https://mermaid.ink/img/pako:eNptkk2L2zAQhv-KmLI3OcjfiQ8L24T0VCgUSttoD7I9zoraUpBliBvy3yspSdeUDsaeGT3zejTSBRrdIlQQRRFXjVadPFZcEWf2DQf8JowUdY_jPentPG_fhLGLjLdTr-1W99p8ET1aixXh8IEFoyQ4-31w9vuQ8R_GOHAVfv1X5-npv1J1L5pflNT9hJQcDaKixGDr689z4_uJarTi3rq0Pbqinew6NKgsabzcSKR6dO8Lw2YicZajY9U01GjG7xzIgcMno6cTeeFAySP4uAy2y2DH4fWmNv-j9sOpMRJFzyRjN8JtL2zlFtXCkEPCaMpokrv36wKacMHEDrg_70wYwwLKqcNymr8TfkLB76VCcvCrmdNwAFAY0AxCtu7wLx7iEE6cgx93pw2OYUhXR4rJ6q-zaqCyxs0fplMrLO6kOBoxQNWJfnTZk1A_tR4ekAuhusAZqmS9KvIkY2mSpmVZlGVJYXbppFzlRZwWWRyXaczWxZXC76DAVpukcNn1pmRpWmzijAK20mrz-XZdw629_gEeXcnm?type=png)


A legend is currently not possible, I've suggested the feature in #5326

**Reply by jeanplevesque** [reply]
Great! thank you!
